src/citrine/informatics/design_candidate.py

```diff
@@ -0,0 +1,104 @@
+from typing import Type
+from citrine._serialization import properties
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization.serializable import Serializable
+
+
+__all__ = [
+    'DesignCandidate',
+    'DesignMaterial',
+    'DesignVariable',
+    'MeanAndStd',
+    'CategoriesAndProbabilities',
+    'Mixture',
+    'ChemicalFormula',
+    'MolecularStructure',
+]
+
+
+class DesignVariable(PolymorphicSerializable["DesignVariable"]):
+    """docstring for DesignVariable"""
+
+    def __init__(self, arg):
+        pass  # pragma: no cover
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        return {
+            "R": MeanAndStd,
+            "C": CategoriesAndProbabilities,
+            "M": Mixture,
+            "F": ChemicalFormula,
+            "S": MolecularStructure
+        }[data["type"]]
+
+
+class MeanAndStd(Serializable["MeanAndStd"], DesignVariable):
+    """docstring for MeanAndStd"""
+
+    mean = properties.Float('m')
+    std = properties.Float('s')
+
+    def __init__(self):
+        pass  # pragma: no cover
+
+
+class CategoriesAndProbabilities(Serializable["CategoriesAndProbabilities"], DesignVariable):
+    """docstring for MeanAndStd"""
+
+    category_probability = properties.Mapping(properties.String, properties.Float, 'cp')
```

Comment:
Is this a map or a pair of coordinated lists in the API?

Reply:
It's a map in the API: `cp: Map[String, Double]` in the DesignResponseObject: https://github.com/CitrineInformatics/platform-backend/blob/c74f9b98d6f8fb6e88252a35eae23c46ba7cc506/code/orion/ml-config-core/src/main/scala/io/citrine/orion/core/domain/DesignResultResponse.scala#L7-L13